enhancements/cluster-logging/elasticsearch-v2-api-and-infrastructure.md

```diff
@@ -0,0 +1,310 @@
+---
+title: elasticsearch-v2-api-and-infrastructure
+authors:
+  - "@ewolinetz"
+reviewers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@alanconway"
+approvers:
+  - "@jcantrill"
+  - "@bparees"
+  - "@alanconway"
+creation-date: 2021-07-02
+last-updated: 2021-06-02
+status: implementable
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# elasticsearch-v2-api-and-infrastructure
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Migration plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently when interacting with the Elasticsearch Operator via the Elasticsearch CRD, one needs to have a lot of knowledge on how to best configure
+the Elasticsearch cluster node's roles, memory/cpu, and replication count. In order to continue to improve the user experience of the operator
+and provide a performant cluster that can operate at scale we should simplify the API and move the knowledge of desired infrastructure into the
+operator itself.
+
+## Motivation
+
+Currently the majority of our Elasticsearch bugs are performance related and are usually clusters that have been scaled vertically rather than horizontally.
+Given each node responsibility of Elasticsearch uses its memory differently, having nodes that act as "All-In-One" causes resources to not be used
+in optimal ways and usually requires an over allocation of resources (which makes it hard to justify horizontal scaling, which leads to performance
+related issues). It would also allow us to move away from the idea of node UUIDs and make naming conventions more generic and simple.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the responsibility for the Elasticsearch Operator regarding desired infrastructure
+
+* Determine a v2 elasticsearch CRD API
+
+* Discuss upgrade strategies from v1 to v2
```

Comment:
Is it really a goal to "discuss" upgrade strategies?  Or is it a goal to "Provide upgrade strategies from v1 to v2".